### PR TITLE
pythonPackages.nmigen, glasgow: update to unstable-2021-02-04

### PR DESCRIPTION
**Pull request by yorickvP**
###### Motivation for this change

The versions in nixpkgs are old and a lot of changes happened in the meantime. Nothing happened to glasgow between 2021-01-09 and 2021-02-06.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nmigen/default.nix [hunk 1]
```diff
@@ -16,21 +17,21 @@
...
-    rev = "c79caead33fff14e2dec42b7e21d571a02526876";
-    sha256 = "sha256-3+mxHyg0a92/BfyePtKT5Hsk+ra+fQzTjCJ2Ech44/s=";
+    rev = "f7c2b9419f9de450be76a0e9cf681931295df65f";
+    sha256 = "0cjs9wgmxa76xqmjhsw4fsb2mhgvd85jgs2mrjxqp6fwp8rlgnl1";
   };
 
   nativeBuildInputs = [ setuptools_scm ];
 
-  propagatedBuildInputs = [ setuptools pyvcd jinja2 ];
+  propagatedBuildInputs = [ setuptools pyvcd jinja2 importlib-resources ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nmigen/default.nix:34]: Please only require this for 3.9 https://github.com/nmigen/nmigen/blob/master/setup.py#L44.

File: pkgs/development/python-modules/nmigen/default.nix [hunk 2]
```diff
@@ -16,21 +17,22 @@
...
-    sha256 = "sha256-3+mxHyg0a92/BfyePtKT5Hsk+ra+fQzTjCJ2Ech44/s=";
+    rev = "f7c2b9419f9de450be76a0e9cf681931295df65f";
+    sha256 = "0cjs9wgmxa76xqmjhsw4fsb2mhgvd85jgs2mrjxqp6fwp8rlgnl1";
   };
 
   nativeBuildInputs = [ setuptools_scm ];
 
-  propagatedBuildInputs = [ setuptools pyvcd jinja2 ];
+  propagatedBuildInputs = [ setuptools pyvcd jinja2 ]
+                          ++ lib.optional (pythonOlder "3.9") importlib-resources;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nmigen/default.nix:35]:
```suggestion
  propagatedBuildInputs = [ setuptools pyvcd jinja2 ]
    ++ lib.optional (pythonOlder "3.9") importlib-resources;
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.nmigen-soc: log <a href="https://termbin.com/zte3">https://termbin.com/zte3</a></li>
<li>python38Packages.nmigen-soc: log <a href="https://termbin.com/ubys">https://termbin.com/ubys</a></li>
<li>python39Packages.nmigen-soc: log <a href="https://termbin.com/okzl">https://termbin.com/okzl</a></li>
  </ul>
</details>
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>glasgow (python38Packages.glasgow)</li>
<li>python37Packages.glasgow</li>
<li>python37Packages.nmigen</li>
<li>python37Packages.nmigen-boards</li>
<li>python38Packages.nmigen</li>
<li>python38Packages.nmigen-boards</li>
<li>python39Packages.glasgow</li>
<li>python39Packages.nmigen</li>
<li>python39Packages.nmigen-boards</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.nmigen-soc: log <a href="https://termbin.com/t1y6">https://termbin.com/t1y6</a></li>
<li>python38Packages.nmigen-soc: log <a href="https://termbin.com/irj9">https://termbin.com/irj9</a></li>
<li>python39Packages.nmigen-soc: log <a href="https://termbin.com/8ybv">https://termbin.com/8ybv</a></li>
  </ul>
</details>
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>glasgow (python38Packages.glasgow)</li>
<li>python37Packages.glasgow</li>
<li>python37Packages.nmigen</li>
<li>python37Packages.nmigen-boards</li>
<li>python38Packages.nmigen</li>
<li>python38Packages.nmigen-boards</li>
<li>python39Packages.glasgow</li>
<li>python39Packages.nmigen</li>
<li>python39Packages.nmigen-boards</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nmigen-boards:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by emilazy** [comment]
Some overlap with (the newer) #112614.

**Pull request closed without merging**